lib/analyse.ml

```diff
@@ -1,14 +1,20 @@
 open Lwt.Infix
 open Current.Syntax
 
-type key = {
-  src : Current_git.Commit.t;
-}
-
 let pool = Current.Pool.create ~label:"analyse" 2
 
+let ( / ) = Filename.concat
 let ( >>!= ) = Lwt_result.bind
 
+let read_file ~max_len path =
+  let ch = open_in path in
+  Fun.protect ~finally:(fun () -> close_in ch)
+    (fun () ->
+       let len = in_channel_length ch in
+       if len <= max_len then really_input_string ch len
```

Comment:
shouldn't we use Lwt functions instead?

Reply:
Could do. Feel free to make a PR! I just copied this code from ocaml-ci, where it doesn't seem to be introducing any noticeable latency.